ui/src/app/utils/search/filter-handlers.ts

```diff
@@ -0,0 +1,230 @@
+export type FilterValue = string | number;
+
+export type Filters = {
+  [x: string]: FilterValue[];
+};
+
+export type PrefixedFilter = {
+  filter: string;
+  prefix: string;
+};
+
+export const isFilterValue = (nodeValue: unknown): nodeValue is FilterValue => {
+  return (
+    (typeof nodeValue === "string" || typeof nodeValue === "number") &&
+    (!!nodeValue || nodeValue === 0)
+  );
+};
+
+export const isFilterValueArray = (
+  nodeValue: unknown
+): nodeValue is FilterValue[] => {
+  return Array.isArray(nodeValue) && isFilterValue(nodeValue[0]);
+};
+
+export default class FilterHandlers {
```

Comment:
I moved these utilities to a class so that the `prefixedFilters` (e.g. for workload annotations) are available to all utilities.